src/backend/api/Fusion.Resources.Domain/Queries/Departments/GetRelevantOrgUnits.cs

```diff
@@ -0,0 +1,382 @@
+                }) ?? Array.Empty<QueryOrgUnitReason>());
+                //retList.AddRange(lineOrgDepartmentProfile?.Siblings.Select(dep => new QueryOrgUnit
+                //{
+                //    FullDepartment = dep.DepartmentId,
+                //    Reason = "RelevantSibling"
+                //}) ?? Array.Empty<QueryOrgUnit>());
+                retList.AddRange(delegatedParentManagerWithResposibility?.Select(dep => new QueryOrgUnitReason
+                {
+                    FullDepartment = dep.DepartmentId,
+                    Reason = "DelegatedParentManager"
+                }) ?? Array.Empty<QueryOrgUnitReason>());
+
+
+                var endResult = new List<QueryRelevantOrgUnit>();
+                foreach (var org in retList)
+                {
+                    if (org?.FullDepartment != null)
+                    {
+                        var alreadyInList = endResult.FirstOrDefault(x => x.FullDepartment == org.FullDepartment);
+
+                        if (alreadyInList is null)
+                        {
+                            var data = cachedOrgUnits.Where(x => x.FullDepartment == org?.FullDepartment.Replace('*', ' ').TrimEnd()).FirstOrDefault();
+
+
+
+
+                            if (data != null)
+                            {
+                                data.Reasons.Clear();
+                                data.Reasons.Add(org.Reason);
+                                endResult.Add(data);
+                            }
+
+                        }
+                        else
+                        {
+                            if (!alreadyInList.Reasons.Contains(org.Reason))
+                            {
+
+                                alreadyInList.Reasons.Add(org.Reason);
+                            }
+
+                        }
+
+
+
+
+                    }
+
+
+                }
+
+                var filteredOrgUnits = ApplyOdataFilters(request.Query, endResult);
+
+                var skip = request.Query.Skip.GetValueOrDefault(0);
+
+                var pagedQuery = QueryRangedList.FromItems(filteredOrgUnits, filteredOrgUnits.Count, skip);
+
+
+                return pagedQuery;
+            }
+
+            private static List<QueryRelevantOrgUnit> ApplyOdataFilters(ODataQueryParams filter, List<QueryRelevantOrgUnit> orgUnits)
+            {
+
+                var filterGenerator = filter.GenerateFilters<QueryRelevantOrgUnit>(m =>
+                {
+                    m.SqlQueryMode = false;
+                    m.MapField("sapId", e => e.SapId);
+                    m.MapField("name", e => e.Name);
+                    m.MapField("shortName", e => e.ShortName);
+                    m.MapField("department", e => e.Department);
+                    m.MapField("fullDepartment", e => e.FullDepartment);
+                    m.MapField("reason", e => e.Reasons);
+                });
+                return orgUnits.Where(filterGenerator.FilterLambda.Compile()).ToList();
```

Comment:
We should probably create some tests evaluating some of these filters.

examples:
$filter=fulldepartment startswith 'MMP'
$filter=shortName contains 'MMP'
$filter=department endswith 'MMP'
$filter=sapId eq '52374964'
$filter=sapId neq '52374964'
$filter=name eq 'People And Organisation'